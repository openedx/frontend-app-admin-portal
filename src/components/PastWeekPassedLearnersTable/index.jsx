import React from 'react';

import TableContainer from '../../containers/TableContainer';
import EnterpriseDataApiService from '../../data/services/EnterpriseDataApiService';
import { formatTimestamp } from '../../utils';

const PastWeekPassedLearnersTable = () => {
  const tableColumns = [
    {
      label: 'Email',
      key: 'user_email',
      columnSortable: true,
    },
    {
      label: 'Course Title',
      key: 'course_title',
      columnSortable: true,
    },
    {
      label: 'Passed Date',
      key: 'passed_date',
      columnSortable: true,
    },
  ];

  const formatLearnerData = learners => learners.map(learner => ({
    ...learner,
    user_email: <span data-hj-suppress>{learner.user_email}</span>,
    passed_date: formatTimestamp({ timestamp: learner.passed_date }),
  }));

  return (
    <TableContainer
      id="completed-learners-week"
      className="completed-learners-week"
      fetchMethod={(enterpriseId, options) => EnterpriseDataApiService.fetchCourseEnrollments(
        enterpriseId,
        {
          passedDate: 'last_week',
          ...options,
        },
      )}
      columns={tableColumns}
      formatData={formatLearnerData}
      tableSortable
    />
  );
};

export default PastWeekPassedLearnersTable;
