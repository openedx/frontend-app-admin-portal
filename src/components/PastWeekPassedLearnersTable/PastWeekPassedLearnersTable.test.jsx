import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

import PastWeekPassedLearnersTable from '.';

const enterpriseId = 'test-enterprise';
const mockStore = configureMockStore([thunk]);
const store = mockStore({
  portalConfiguration: {
    enterpriseId,
  },
  table: {
    'completed-learners-week': {
      data: {
        count: 2,
        num_pages: 1,
        current_page: 1,
        results: [
          {
            id: 1,
            passed_date: '2018-09-23T16:27:34.690065Z',
            course_title: 'Dive into ReactJS',
            course_key: 'edX/ReactJS',
            user_email: 'awesome.me@example.com',
          },
          {
            id: 5,
            passed_date: '2018-09-22T16:27:34.690065Z',
            course_title: 'Redux with ReactJS',
            course_key: 'edX/Redux_ReactJS',
            user_email: 'new@example.com',

          },
        ],
        next: null,
        start: 0,
        previous: null,
      },
      ordering: null,
      loading: false,
      error: null,
    },
  },
});

const PastWeekPassedLearnersWrapper = props => (
  <MemoryRouter>
    <Provider store={store}>
      <PastWeekPassedLearnersTable
        {...props}
      />
    </Provider>
  </MemoryRouter>
);

describe('PastWeekPassedLearnersTable', () => {
  let wrapper;

  it('renders table correctly', () => {
    const { container: tree } = render(
      <PastWeekPassedLearnersWrapper />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders table with correct data', () => {
    const tableId = 'completed-learners-week';
    const columnTitles = ['Email', 'Course Title', 'Passed Date'];
    const rowsData = [
      [
        'awesome.me@example.com',
        'Dive into ReactJS',
        'September 23, 2018',
      ],
      [
        'new@example.com',
        'Redux with ReactJS',
        'September 22, 2018',
      ],
    ];

    wrapper = render((
      <PastWeekPassedLearnersWrapper />
    ));

    // Verify that table has correct number of columns
    expect(wrapper.container.querySelectorAll(`.${tableId} thead th`).length).toEqual(3);

    // Verify only expected columns are shown
    wrapper.container.querySelectorAll(`.${tableId} thead th`).forEach((column, index) => {
      expect(column.textContent).toContain(columnTitles[index]);
    });

    // Verify that table has correct number of rows
    expect(wrapper.container.querySelectorAll(`.${tableId} tbody tr`).length).toEqual(2);

    // Verify each row in table has correct data
    wrapper.container.querySelectorAll(`.${tableId} tbody tr`).forEach((row, rowIndex) => {
      row.querySelectorAll('td').forEach((cell, colIndex) => {
        expect(cell.textContent).toEqual(rowsData[rowIndex][colIndex]);
      });
    });
  });
});
