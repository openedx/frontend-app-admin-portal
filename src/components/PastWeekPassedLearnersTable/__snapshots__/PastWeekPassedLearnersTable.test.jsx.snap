// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PastWeekPassedLearnersTable renders table correctly 1`] = `
<div
  className="completed-learners-week"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="table-responsive"
      >
        <table
          className="table table-sm table-striped"
        >
          <thead
            className=""
          >
            <tr
              className=""
            >
              <th
                className="sortable"
                scope="col"
              >
                <button
                  className="btn btn-header"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span>
                    Email
                    <span
                      className="sr-only"
                    >
                       
                      sort ascending
                    </span>
                     
                    <span
                      aria-hidden={true}
                      className="fa fa-sort-asc"
                    />
                  </span>
                </button>
              </th>
              <th
                className="sortable"
                scope="col"
              >
                <button
                  className="btn btn-header"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span>
                    Course Title
                    <span
                      className="sr-only"
                    >
                       
                      click to sort
                    </span>
                     
                    <span
                      aria-hidden={true}
                      className="fa fa-sort"
                    />
                  </span>
                </button>
              </th>
              <th
                className="sortable"
                scope="col"
              >
                <button
                  className="btn btn-header"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span>
                    Passed Date
                    <span
                      className="sr-only"
                    >
                       
                      click to sort
                    </span>
                     
                    <span
                      aria-hidden={true}
                      className="fa fa-sort"
                    />
                  </span>
                </button>
              </th>
            </tr>
          </thead>
          <tbody
            className=""
          >
            <tr
              className=""
            >
              <td
                className=""
                scope={null}
              >
                <span
                  data-hj-suppress={true}
                >
                  awesome.me@example.com
                </span>
              </td>
              <td
                className=""
                scope={null}
              >
                Dive into ReactJS
              </td>
              <td
                className=""
                scope={null}
              >
                September 23, 2018
              </td>
            </tr>
            <tr
              className=""
            >
              <td
                className=""
                scope={null}
              >
                <span
                  data-hj-suppress={true}
                >
                  new@example.com
                </span>
              </td>
              <td
                className=""
                scope={null}
              >
                Redux with ReactJS
              </td>
              <td
                className=""
                scope={null}
              >
                September 22, 2018
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
  <div
    className="row mt-2"
  >
    <div
      className="col d-flex justify-content-center"
    >
      <nav
        aria-label="completed-learners-week-pagination"
        className="pagination-default"
      >
        <div
          aria-atomic={true}
          aria-live="polite"
          aria-relevant="text"
          className="sr-only"
        >
          Page 1, Current Page, of 1
        </div>
        <ul
          className="pagination"
        >
          <li
            className="page-item disabled"
          >
            <button
              aria-label="Previous"
              className="previous page-link btn btn-primary"
              disabled={true}
              onClick={[Function]}
              tabIndex="-1"
              type="button"
            >
              <div>
                <span
                  className="pgn__icon"
                >
                  <svg
                    aria-hidden={true}
                    fill="none"
                    focusable={false}
                    height={24}
                    role="img"
                    viewBox="0 0 24 24"
                    width={24}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12l4.58-4.59Z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
                Previous
              </div>
            </button>
          </li>
          <li
            className="page-item disabled"
          >
            <button
              aria-label="Next"
              className="next page-link btn btn-primary"
              disabled={true}
              onClick={[Function]}
              tabIndex="-1"
              type="button"
            >
              <div>
                Next
                <span
                  className="pgn__icon"
                >
                  <svg
                    aria-hidden={true}
                    fill="none"
                    focusable={false}
                    height={24}
                    role="img"
                    viewBox="0 0 24 24"
                    width={24}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6-6-6Z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </div>
            </button>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</div>
`;
